maggy/util.py

```diff
@@ -234,3 +229,48 @@ def _validate_ml_id(app_id, run_id):
     if prev_app_id == app_id and int(prev_run_id) >= run_id:
         return app_id, (int(prev_run_id) + 1)
     return app_id, run_id
+
+
+def _find_spark(conf=None):
+    """
+    Returns: SparkSession
+    """
+    # sp = SparkSession.builder.getOrCreate()
+    # sp.stop()
+
+    conf = SparkConf()
+    conf.set("num-executors", "1")
+
+    return (
+        SparkSession.builder.getOrCreate()
+        if not conf
+        else SparkSession.builder.config(conf=conf).getOrCreate()
+    )
+
+
+def _seconds_to_milliseconds(time):
+    """
+    Returns: time converted from seconds to milliseconds
+    """
+    return int(round(time * 1000))
+
+
+def _time_diff(t0, t1):
+    """
+    Args:
+        :t0: start time in seconds
+        :t1: end time in seconds
+
+    Returns: string with time difference (i.e. t1-t0)
+
+    """
+
+    millis = _seconds_to_milliseconds(t1) - _seconds_to_milliseconds(t0)
+    millis = int(millis)
+    seconds = (millis / 1000) % 60
+    seconds = int(seconds)
+    minutes = (millis / (1000 * 60)) % 60
+    minutes = int(minutes)
+    hours = (millis / (1000 * 60 * 60)) % 24
+
+    return "%d hours, %d minutes, %d seconds" % (hours, minutes, seconds)
```

Comment:
Nit: We should talk to Moritz about the leader lowercase namestyle of the methods in util. It's a util, it's supposed to be imported and used outside the file. Private naming convention for these functions makes no sense. You could start with the newly introduced functions, but in general, all functions in util should use the public naming convention if we want to use them outside util.

Reply:
yes agree, I copied them at some point from hops-util-py where they were private. feel free to make em public